connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginsTest.java

```diff
@@ -185,6 +181,102 @@ public void shouldInstantiateAndConfigureDefaultHeaderConverter() {
         assertTrue(headerConverter instanceof SimpleHeaderConverter);
     }
 
+    @Test(expected = ConnectException.class)
+    public void shouldThrowIfPluginThrows() {
+        TestPlugins.assertInitialized();
+
+        plugins.newPlugin(
+            TestPlugins.ALWAYS_THROW_EXCEPTION,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        );
+    }
+
+    @Test
+    public void shouldShareStaticValuesBetweenSamePlugin() {
+        // Plugins are not isolated from other instances of their own class.
+        TestPlugins.assertInitialized();
+        Converter firstPlugin = plugins.newPlugin(
+            TestPlugins.SAMPLING,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        );
+
+        assertTrue(firstPlugin instanceof SamplingTestPlugin);
+
+        assertEquals(1, ((SamplingTestPlugin) firstPlugin).dynamicInitializations());
+
+        Converter secondPlugin = plugins.newPlugin(
+            TestPlugins.SAMPLING,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        );
+
+        assertTrue(secondPlugin instanceof SamplingTestPlugin);
+
+        assertEquals(2, ((SamplingTestPlugin) secondPlugin).dynamicInitializations());
+
+        // This value changes because secondPlugin's instantiation incremented it.
+        assertEquals(2, ((SamplingTestPlugin) firstPlugin).dynamicInitializations());
+    }
+
+    @Test
+    public void shouldPerformServiceLoadingWithPluginClassloader() {
+        TestPlugins.assertInitialized();
+        Converter plugin = plugins.newPlugin(
+            TestPlugins.SERVICE_LOADER,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        );
+
+        assertTrue(plugin instanceof SamplingTestPlugin);
+        SamplingTestPlugin samplingPlugin = (SamplingTestPlugin) plugin;
+        for (Entry<String, SamplingTestPlugin> e : samplingPlugin.otherSamples().entrySet()) {
+            String message = e.getKey() + "was not initialized properly";
+            SamplingTestPlugin sample = e.getValue();
+            assertTrue(message, sample.staticClassloader() instanceof PluginClassLoader);
+            assertTrue(message, sample.classloader() instanceof PluginClassLoader);
+            assertEquals(message, sample.staticClassloader(), sample.classloader());
+        }
+        Map<String, SamplingTestPlugin> samples = samplingPlugin.otherSamples();
+        // Initialized once per (static, dynamic, static subclass, dynamic subclass)
+        assertEquals(
+            4,
```

Comment:
same comment about expected numbers applies to all assertions and possibly other test cases.

Reply:
This instance is already covered by the comment above.